File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const seen = {};

        for (const num of nums) {
            if (seen[num]) {
                return true
            } else {
                seen[num] = true;
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        bucket = [[] for i in range(len(nums) + 1)]

        for num in nums:
            if num not in freq:
                freq[num] = 1
            else:
                freq[num] += 1

        for key, val in freq.items():
            bucket[val].append(key)

        mostFreq = []
        for i in range(len(bucket) - 1, 0, -1):
            for n in bucket[i]:
                mostFreq.append(n)
                if len(mostFreq) == k:
                    return mostFreq
